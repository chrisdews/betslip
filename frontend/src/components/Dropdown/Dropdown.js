import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

function Dropdown({ setOddsFilter }) {
  const handleSelect = (e) => {
    setOddsFilter(e.target.value);
  };

  return (
    <DropdownStyle name="selectList" id="selectList" onChange={handleSelect}>
      <OptionStyle value="all">ALL</OptionStyle>
      <OptionStyle value="decimalOddsLessThanTwo">Odds less than 2</OptionStyle>
      <OptionStyle value="decimalOddsMoreThanTwo">
        Odds greater than 2
      </OptionStyle>
    </DropdownStyle>
  );
}

Dropdown.propTypes = {
  setOddsFilter: PropTypes.func,
};

export default Dropdown;

const DropdownStyle = styled.select`
  border-radius: 0.5em;
  border: 2px solid white;
  margin: auto;
  margin-right: 1em;
  font-size: 100%;
  font: inherit;
  width: 8em;
  color: white;
  background-color: inherit;
  padding: 0.5em;
`;

const OptionStyle = styled.option`
  font-size: 100%;
  font: inherit;
  width: 8em;
  color: white;
  background-color: inherit;
  padding: 0.5em;
`;
