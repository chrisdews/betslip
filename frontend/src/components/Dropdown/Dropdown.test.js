import React from "react";
import Dropdown from "./Dropdown";
import { render, fireEvent } from "@testing-library/react";

describe("Dropdown", () => {
    it("renders correctly - snapshot", () => {
      const { container } = render(<Dropdown />);
      expect(container).toMatchSnapshot();
    });
  
    it("should run the setOddsFilter function when each option is selected", () => {
      const filterFunction = jest.fn();
      const { getByTestId } = render(
        <Dropdown setOddsFilter={filterFunction} />
      );
     
      fireEvent.change(getByTestId("dropdown-select"), {
        target: { value: "decimalOddsLessThanTwo" },
      });
      expect(filterFunction).toBeCalledWith("decimalOddsLessThanTwo");

      fireEvent.change(getByTestId("dropdown-select"), {
        target: { value: "decimalOddsMoreThanTwo" },
      });
      expect(filterFunction).toBeCalledWith("decimalOddsMoreThanTwo");

      fireEvent.change(getByTestId("dropdown-select"), {
        target: { value: "all" },
      });
      expect(filterFunction).toBeCalledWith("all");
    });
  });