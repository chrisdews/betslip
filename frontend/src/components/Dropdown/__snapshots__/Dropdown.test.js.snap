// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly - snapshot 1`] = `
<div>
  <select
    class="sc-bdnxRM einyrw"
    data-testid="dropdown-select"
    id="selectList"
    name="selectList"
  >
    <option
      class="sc-gtsrHT DFeHs"
      value="all"
    >
      ALL
    </option>
    <option
      class="sc-gtsrHT DFeHs"
      value="decimalOddsLessThanTwo"
    >
      Odds less than 2
    </option>
    <option
      class="sc-gtsrHT DFeHs"
      value="decimalOddsMoreThanTwo"
    >
      Odds greater than 2
    </option>
  </select>
</div>
`;
